IYermakov/Snakes
# Jenkinsfile has been updated

## Github Issue #
Added block with parameters;
Added stage "Tagging" to make Releases in both repos;
Added ability to choose which stage to perform;
Added functionality to auto-increase the version of the application;
Added functionality to choose the amount of traffic to the new version of the App.

When 'SetNewTag' is selected, Jenkins will automatically increase the version of the Tag, depending on the option selected.
When 'Release' is selected, Jenkins will create a new Tag-based release and push it into the App repo and OPS-repo.
When 'Deployment' is selected, Jenkins will create a new CFN Stack in AWS and deploy a new version of the application in ECS. Part of the traffic will be redirected to the new version if select a value of "NewVersionTrafficWeight" greater than 0.

## What Are We Doing Here?

Changes in the Jenkinsfile allow you to run a parameterized build, with the ability to select the stages of execution. Also added the ability to determine: how much traffic redirect to the new version of the App.


## How to Test and Verify

1. Check out this PR
2. Go to the https://ci.devopsa3.me.uk/job/Snakes-deploy-new-version/
3. Click 'Build with Parameters'
4. Choose options and click 'Build'

## Risk

### Level

`High`

### Required Testing
`Manual`

### Risk Summary

If you choose the number closer to 10 (for example: '9' ) in "NewVersionTrafficWeight" parameter, then 90% of the traffic will be redirected to the new version of the App, and only 10% will be sent to the current version.

## Code Review Checklist


we inevitably come back to this code it will only take hours to figure out, not
days.
